src/front-door/azext_front_door/custom.py

```diff
@@ -454,6 +454,39 @@ def add_fd_backend(cmd, resource_group_name, front_door_name, backend_pool_name,
     return backend
 
 
+def update_fd_backend(cmd, resource_group_name, front_door_name, backend_pool_name, index, address=None,
+                      http_port=None, https_port=None, disabled=None, priority=None, weight=None,
+                      backend_host_header=None, private_link_alias=None, private_link_resource_id=None,
+                      private_link_location=None, private_link_approval_message=None):
+
+    client = cf_frontdoor(cmd.cli_ctx, None)
+    frontdoor = client.get(resource_group_name, front_door_name)
+    backend_pool = next((x for x in frontdoor.backend_pools if x.name == backend_pool_name), None)
+    if not backend_pool:
+        from knack.util import CLIError
+        raise CLIError("Backend pool '{}' could not be found on frontdoor '{}'".format(
```

Comment:
`CLIError` is not used. Please follow the guideline https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md